cirq-core/cirq/transformers/routing/mapping_manager_test.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import networkx as nx
+from networkx.utils.misc import graphs_equal
+
+import cirq
+
+
+def test_mapping_manager():
```

Comment:
Can we break this down into multiple smaller unit tests, each of which test a particular behavior / interface? The logic to construct the `device_graph` and `initial_mapping` can be shared by all unit tests using either fixtures or calling a global method `def construct_device_graph_and_mapping`.